### PR TITLE
introduce unnamed slot behaviour

### PR DESCRIPTION
**Pull request by vigneshshanmugam**
This will allow us to insert contents anywhere in the parent template without ugly slot name magic in the child templates.

### CODE REVIEW

File: lib/serializer.js [hunk 1]
```diff
@@ -77,11 +78,18 @@ module.exports = class CustomSerializer extends Serializer {
...
-        if (this.slotMap.has(slotName)) {
+        if (slotName && this.slotMap.has(slotName)) {
             const slots = this.slotMap.get(slotName);
             if (slots) {
                 slots.forEach(this._serializeNode);
             }
+        } else {
+            // Unnamed slot behaviour
+            const defaultSlots = this.slotMap.get('default');
+            this.defaultSlotsInserted = true;
```
Comment (grassator) [inline lib/serializer.js]: We should check what does the spec say about duplicate default slots. Maybe we should also warn on that.

File: lib/serializer.js [hunk 2]
```diff
@@ -77,11 +78,18 @@ module.exports = class CustomSerializer extends Serializer {
...
         const slotName = node.attribs.name;
-        if (this.slotMap.has(slotName)) {
+        if (slotName && this.slotMap.has(slotName)) {
             const slots = this.slotMap.get(slotName);
             if (slots) {
                 slots.forEach(this._serializeNode);
             }
+        } else {
+            // Unnamed slot behaviour
+            const defaultSlots = this.slotMap.get('default');
```
Comment (grassator) [inline lib/serializer.js]: Where do you "set" that unnamed slot as a "default"?

Comment (vigneshshanmugam) [inline lib/serializer.js]: [Here](https://github.com/zalando/tailor/blob/master/lib/transform.js#L33) i am grouping them together to default slots if they don't have a slot name

File: lib/serializer.js (hunk 1 above)
Comment (vigneshshanmugam) [inline lib/serializer.js]: checked the spec, Sadly nowhere it was mentioned

**Comment by grassator** [comment]
👍

**Pull request merged**